examples/pipeline_wavernn/wavernn.py

```diff
@@ -0,0 +1,586 @@
+                raise ValueError(
+                    f"Expected mode: `waveform` or `mol`, but found {mode}"
+                )
+
+            loss = criterion(output, target)
+            sums["loss"] += loss.item()
+
+            if SIGNAL_RECEIVED:
+                break
+
+        avg_loss = sums["loss"] / len(data_loader)
+
+        metric = MetricLogger("validation")
+        metric("epoch", epoch)
+        metric("loss", avg_loss)
+        metric("time", time() - start)
+        metric.print()
+
+        return avg_loss
+
+
+def main(args):
+
+    devices = ["cuda" if torch.cuda.is_available() else "cpu"]
+
+    logging.info("Start time: {}".format(str(datetime.now())))
+
+    # Empty CUDA cache
+    torch.cuda.empty_cache()
+
+    # Install signal handler
+    signal.signal(signal.SIGUSR1, lambda a, b: signal_handler(a, b))
+
+    # use torchaudio transform to get waveform and specgram
+
+#    melkwargs = {
+#        "n_fft": 2048,
+#        "n_mels": args.n_freq,
+#        "hop_length": args.hop_length,
+#        "f_min": args.f_min,
+#        "win_length": args.win_length
+#    }
+
+#    transforms = torch.nn.Sequential(
+#        torchaudio.transforms.MelSpectrogram(
+#            sample_rate=args.sample_rate, **melkwargs
+#        ),
+#    )
+
+    # use librosa transform to get waveform and specgram
+
+    melkwargs = {
+        "n_fft": 2048,
+        "num_mels": args.n_freq,
+        "hop_length": args.hop_length,
+        "fmin": args.f_min,
+        "win_length": args.win_length,
+        "sample_rate": args.sample_rate,
+        "min_level_db": args.min_level_db
+    }
+    transforms = Transform(**melkwargs)
+
+    # dataset
+    train_dataset, test_dataset = datasets_ljspeech(args, transforms)
+
+    loader_training_params = {
+        "num_workers": args.workers,
+        "pin_memory": False,
+        "shuffle": True,
+        "drop_last": False,
+    }
+    loader_validation_params = loader_training_params.copy()
+    loader_validation_params["shuffle"] = False
+
+    collate_fn = collate_factory(args)
+
+    loader_training = DataLoader(
+        train_dataset,
+        batch_size=args.batch_size,
+        collate_fn=collate_fn,
+        **loader_training_params,
+    )
+    loader_test = DataLoader(
+        test_dataset,
+        batch_size=args.batch_size,
+        collate_fn=collate_fn,
+        **loader_validation_params,
+    )
+
+    # model
+    model = _WaveRNN(upsample_scales=args.upsample_scales,
+                     n_bits=args.n_bits,
+                     sample_rate=args.sample_rate,
+                     hop_length=args.hop_length,
+                     n_res_block=args.n_res_block,
+                     n_rnn=args.n_rnn,
+                     n_fc=args.n_fc,
+                     kernel_size=args.kernel_size,
+                     n_freq=args.n_freq,
+                     n_hidden=args.n_hidden,
+                     n_output=args.n_output,
+                     mode=args.mode,
+                     )
```

Comment:
I don't remember what's the standard for `black`. Have you ran `black` on this file?

Reply:
Yes. I ran `black'.